autoload/codi.vim

```diff
@@ -693,3 +694,9 @@ function! codi#run(bang, ...)
     return s:codi_spawn(filetype)
   endif
 endfunction
+
+function! codi#log(message)
+    if g:codi#log_filepath != ''
+      call writefile([a:message], g:codi#log_filepath, 'a')
```

Comment:
It may also be helpful to get the timestamp using `strftime` and attach it to the log message.